specs/waku/waku-rpc-api.md

```diff
@@ -0,0 +1,410 @@
+
+## Wire Protocol
+
+### Transport
+
+Nodes SHOULD expose a [JSON RPC](https://www.jsonrpc.org/specification) API that can be publicly accessed. The JSON RPC version SHOULD be `2.0`. Below is an example request:
+
+```json
+{
+  "jsonrpc":"2.0",
+  "method":"shh_version",
+  "params":[],
+  "id":1
+}
+```
+
+#### Fields
+
+| Field     | Description                                         |
+| --------- | --------------------------------------------------- |
+| `jsonrpc` | Contains the used JSON RPC version (`Default: 2.0`) |
+| `method`  | Contains the JSON RPC method that is being called   |
+| `params`  | An array of parameters for the request              |
+| `id`      | The request ID                                      |
+
+### Objects
+
+In this section you will find objects used throughout the JSON RPC API.
+
+#### Message
+
+The message object represents a Waku message. Below you will find the description of the attributes contained in the message object.
+
+| Field | Type | Description |
+| ----: | :--: | ----------- |
+| `sig` | string | Public Key that signed this message |
+| `recipientPublicKey` | string | The recipients public key |
+| `ttl` | number | Time-to-live in seconds |
+| `timestamp` | number | Unix timestamp of the message generation |
+| `topic` | string | 4 bytes, the message topic |
+| `payload` | string | Decrypted payload |
+| `padding` | string | Optional padding, byte array of arbitrary length |
+| `pow` | number | The proof of work value |
+| `hash` | string | Hash of the enveloped message |
+
+#### Filter
+
+The filter object represents filters that can be applied to retrieve messages. Below you will find the description of the attributes contained in the filter object.
+
+| Field | Type | Description |
+| ----: | :--: | ----------- |
+| `symKeyID` | string | ID of the symmetric key for message decryption |
+| `privateKeyID` | string | ID of private (asymmetric) key for message decryption |
+| `sig` | string | Public key of the signature |
+| `minPow` | number | Minimal PoW requirement for incoming messages |
+| `topics` | array | Array of possible topics, this can also contain partial topics |
+| `allowP2P` | boolean | Indicates if this filter allows processing of direct peer-to-peer messages |
+
+All fields are optional, however `symKeyID` or `privateKeyID` must be present, it cannot be both. Additionally, the `topics` field is only optional when an asymmetric key is used.
+
+### Methods
+
+#### `waku_version`
+
+The `waku_version` method returns the current version number.
+
+##### Parameters
+
+none
+
+##### Response
+
+- **string** - The version number.
+
+#### `waku_info`
+
+The `waku_info` method returns information about a Waku node.
+
+##### Parameters
+
+none
+
+##### Response
+
+The response is an `Object` containing the following fields:
+
+- **`minPow` [number]** - The current PoW requirement.
+- **`maxMessageSize` [float]** - The current maximum message size in bytes.
```

Comment:
Considering comment below, perhaps we should consider renaming this also as message size check is done on envelope size.